setup.py

```diff
@@ -124,13 +124,29 @@ def initialize_options(self) -> None:
     'nose'
 ]
 
+smart_open_dependencies = [
+    # we rely on exception types from smart_open,
+    # which seem to change in feature releases
+    # without a major version bump
+    'smart_open>=2,<2.1',
+]
+
+aws_dependencies = [
+    'awscli>=1,<2',
+    'boto>=2,<3',
+    'boto3',
+    's3-concat>=0.1.7,<0.2'
+] + smart_open_dependencies
+
 itest_dependencies = [
     'jsonschema',  # needed for directory_validator.py
 ] + (
     nose_dependencies +
     # needed for records_database_fixture retrying drop/creates on
     # BigQuery
-    google_api_client_dependencies
+    google_api_client_dependencies +
+    # database tests include loading from an S3 bucket
+    aws_dependencies
```

Comment:
Change 1: This ensures we always have AWS dependencies installed when running integration tests.  I have an integration test which tests a database importing from S3--I wanted to be able to test that scenario with more databases rather than skip the test because libraries aren't installed.